aries_vcx_core/src/errors/mapping_indyvdr_proxy.rs

```diff
@@ -0,0 +1,12 @@
+use indy_vdr_proxy_client::error::VdrProxyClientError;
+
+use super::error::{AriesVcxCoreError, AriesVcxCoreErrorKind};
+
+impl From<VdrProxyClientError> for AriesVcxCoreError {
```

Comment:
Currently we have 3 different ledger errors
```
#[error("Connection to Pool Ledger.")]
    PoolLedgerConnect,
    #[error("Ledger rejected submitted request.")]
    InvalidLedgerResponse,
    #[error("Ledger item not found.")]
    LedgerItemNotFound,
```

when using `VdrProxy` ledger trait implementation, all errors are squashed into `AriesVcxCoreErrorKind::InvalidLedgerResponse,` - that's fine with the first two, but remapping `LedgerItemNotFound` into `InvalidLedgerResponse` can cause some troubles. For example function `_try_get_cred_def_from_ledger` is handling `LedgerItemNotFound` in very particular way. 

It would be good if we could on level of `indy-vdr-proxy-client` distinquish and return "item not not found on ledger" vs "other issue with ledger response" errors, and that would enable us propagate more granular handling on aries-vcx level as well

Reply:
Just realized this is actually a problem of wider scope - `vdr_ledger` submitter, eg previously `indy_vdr_ledger.rs` also behaves differently 
```
let send_req_result: VdrSendRequestResult = recv
            .await
            .map_err(|e| AriesVcxCoreError::from_msg(AriesVcxCoreErrorKind::InvalidState, e))?;
        let (result, _) = send_req_result?;

```
It actually returns even different error.

I think you can still line up the error handling as suggested in previous comment, but I think what needs to happen in the end is:

- take out the ledger trait as separate crate
- the trait needs to return custom error type - which could be for example be enum with variants `PoolLedgerConnect`, `InvalidLedgerResponse`, `LedgerItemNotFound`
- Add some documentation to traits specifying what error needs to be returned in what case, ideally covered with some tests against the trait interface which any implementor of the trait could reuse (not sure how this would be practically done)

This should solve the deviations in error handling of the trait implementation, which now returns `AriesVcxCoreError` - way too much freedom for trait implementors.